src/magic_folder/web.py

```diff
@@ -98,17 +98,45 @@ def getChildWithDefault(self, path, request):
 
 
 @attr.s
-class V1MagicFolderAPI(Resource, object):
+class APIv1(Resource, object):
     """
-    The root of the ``/v1`` HTTP API hierarchy.
+    Implement the ``/v1`` HTTP API hierarchy.
 
-    :ivar (unicode -> MagicFolder) _get_magic_folder: A function that looks up
-        a magic folder by its nickname.
+    :ivar MagicFolderServiceState _magic_folder_state: The Magic Folder state
+        to serve.
     """
-    _get_magic_folder = attr.ib()
+    _magic_folder_state = attr.ib()
 
     def __attrs_post_init__(self):
         Resource.__init__(self)
+        self.putChild(b"magic-folder", MagicFolderAPIv1(self._magic_folder_state))
+
+
+@attr.s
+class MagicFolderAPIv1(Resource, object):
+    """
+    Implement the ``/v1/magic-folder`` HTTP API hierarchy.
+
+    :ivar MagicFolderServiceState _magic_folder_state: The Magic Folder state
+        to serve.
+    """
+    _magic_folder_state = attr.ib()
+
+    def __attrs_post_init__(self):
+        Resource.__init__(self)
+
+    def render_GET(self, request):
+        """
+        Render a list of Magic Folders and some of their details, encoded as JSON.
+        """
+        request.responseHeaders.setRawHeaders(u"content-type", [u"application/json"])
+        return json.dumps({
+            u"folders": list(
+                {u"name": name, u"local-path": config[u"directory"]}
+                for (name, config)
+                in sorted(self._magic_folder_state.iter_magic_folder_configs())
+            ),
+        })
```

Comment:
"In general" I like adding `indent=4` or similar to get slightly-more-human-readable output; costs a few more bytes but might make debugging easier (for integrations too, not just us). I guess browsers do formatting too, though, and there's `jq` or others for cli users....

Reply:
Maybe some `magic_folder._json.dumps` could encode this policy for all JSON use throughout the project?